torch/_inductor/compile_fx.py

```diff
@@ -1838,3 +1838,79 @@ def warn_and_skip(device: Optional[torch.device]) -> Never:
         if device_interface.is_bf16_supported(including_emulation=False):
             return
         warn_and_skip(node.get_device())
+
+
+def _flatten_inputs(
```

Comment:
I understand this is moved from _inductor/__init__.py, but I feel flattening inputs should be a functionality provided by Export. @angelayi , does this already exist in Export?

Reply:
(Angela pls correct me if I'm wrong) My understanding is that export does flatten the inputs, but that's the inputs for the "flattened graph" (`exported_program.graph`), but this graph might be different from the input graph module to AOTI (`exported_program.module()`) .

The main difference is that, `exported_program.module()` may contain submodules, but `exported_program.graph` is fully flatten. 

So the two flattening are a little different. Here in _flatten_inputs, we are only flattening the inputs, but not other parts of the graph (e.g. the submodules).

Reply:
Yeah this code is pretty similar to what we have [here](https://github.com/pytorch/pytorch/blob/main/torch/export/_unlift.py#L23), but I'm not sure if it's easy to refactor things to reuse this.

Reply:
@desertfire The `_flatten_inputs` function here does 3 things:

1. update the `gm`'s codegen to take in the flattened inputs
2. get the flattened inputs (this is similar to the code that Angela linked above, but that's basically 2 lines)
3. update the inductor_config's `serialized_in_spec` and `serialized_out_spec` fields

Maybe I could rename this function to e.g. `_update_gm_input_spec_and_flatten` to be more clear that this function does more than just getting the flattened inputs?

Reply:
Or you rename to something more generic but with `aoti` in it.

Reply:
> Or you rename to something more generic but with `aoti` in it.

sure, removed to `_aoti_flatten_inputs` now.